docs/learning/frontend/Extensions.md

```diff
@@ -0,0 +1,196 @@
+# iTwin.js Extensions
+
+An iTwin.js Extension is a separate JavaScript module that can load on demand into an iTwin.js frontend application.
+The separate deliverable enables Extensions to provide extensibility of an iTwin.js application at runtime.
+Extensions have access to a limited set of iTwin.js functionality through @itwin/core-extension to enable seamless integration with the host app.
+
+## What can extensions do?
+
+Extensions can be used for many different purposes, such as:
+- Add new a new tool or marker to an existing application to better support your custom workflows.
```

Comment:
I meant marker because that is the most common use case I've seen - and the only decorator I've actually tested - but you are right, I don't think there's a reason to limit to just a marker.

Reply:
which term is more meaningful to users? Asking because I honestly don't know